File: api/src/main/java/gg/mineral/api/command/Command.java
package gg.mineral.api.command;

import lombok.Data;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Data
public abstract class Command {
    private final String permission, name;

    /**
     * Executes the command.
     * 
     * @param commandExecutor The command executor.
     * @param arguments       The arguments.
     */
    public abstract void execute(CommandExecutor commandExecutor, String[] arguments);
}

File: api/src/main/java/gg/mineral/api/command/Command.kt
package gg.mineral.api.command

import lombok.Data
import lombok.RequiredArgsConstructor

@RequiredArgsConstructor
@Data
abstract class Command {
    private val permission: String? = null
    private val name: String? = null

    /**
     * Executes the command.
     *
     * @param commandExecutor The command executor.
     * @param arguments       The arguments.
     */
    abstract fun execute(commandExecutor: CommandExecutor?, arguments: Array<String?>?)
}
